### PR TITLE
Feature/map redux actions

### PR DESCRIPTION
**Pull request by GiftLanga**

### CODE REVIEW

File: waterlog/src/Components/Map.js [hunk 1]
```diff
@@ -27,7 +30,95 @@ const backgroundColor = '#253238';
...
+  markers = data.monitors.map(mon => {
+    return { id: mon.id, lat: mon.lat, lon: mon.long, status: mon.status }
+  });
+  return { markers, segments };
+}
+
+function generateMapIcons({ segments, markers },  simpleView) {
+  let defaultColor = simpleView ? backgroundColor : lighterColor;
+  return segments.map((segment) => {
+    let sensorIn = markers.find(marker => (marker.id === segment.senseIDIn));
```
Comment (FredJEhlers) [inline waterlog/src/Components/Map.js]: If you are not re-assigning then please use const. goes for the other variables as well

File: waterlog/src/Components/Map.js [hunk 2]
```diff
@@ -41,8 +132,21 @@ export default class MapComponent extends Component {
 	}
 
 	render() {
+    const { error, loading, mapData } = this.props;
+    if (error) {
+			return <div>Error! {error.message}</div>;
```
Comment (FredJEhlers) [inline waterlog/src/Components/Map.js]: indentation and remove console log

File: waterlog/src/Components/Map.js [hunk 3]
```diff
@@ -130,3 +180,17 @@ export default class MapComponent extends Component {
...
 }
+
+
+MapComponent.propTypes = {
+	fetchMapsData: PropTypes.func.isRequired,
+	mapData: PropTypes.array.isRequired
+};
+
+const mapStateToProps = (state) => ({
+  mapData: state.maps.items,
```
Comment (FredJEhlers) [inline waterlog/src/Components/Map.js:190]: indentation is inconsistent

File: waterlog/src/reducers/index.js [hunk 4]
```diff
@@ -31,5 +32,6 @@ export default combineReducers({
 	seasonUsage: UsageGraphSeasonallyReducer,
 	levels: TankLevelReducer,
 	level: TankLevelsByIdReducer,
-   	pumps: PumpsReducer
+  pumps: PumpsReducer,
```
Comment (FredJEhlers) [inline waterlog/src/reducers/index.js:41]: indentation

File: waterlog/src/Components/Map.js (hunk 1 above)
Comment (GiftLanga) [reply waterlog/src/Components/Map.js]: I changed all my variables that are not re-assigned to const

File: waterlog/src/Components/Map.js (hunk 2 above)
Comment (GiftLanga) [reply waterlog/src/Components/Map.js]: I removed console log, and fixed indentation

File: waterlog/src/Components/Map.js (hunk 3 above)
Comment (GiftLanga) [reply waterlog/src/Components/Map.js:190]: fixed indentation

File: waterlog/src/reducers/index.js (hunk 4 above)
Comment (GiftLanga) [reply waterlog/src/reducers/index.js:41]: fixed indentation

File: waterlog/src/Components/Map.js [hunk 5]
```diff
@@ -1,132 +1,189 @@
...
-							if (sensorIn.status.toLowerCase() === 'fault') {
-								sensorInColor = errorColor;
-							}
-							if (sensorOut.status.toLowerCase() === 'fault') {
-								sensorOutColor = errorColor;
-							}
-							if (segment.status.toLowerCase() === 'leak') {
-								segmentColor = errorColor;
-							}
+    if (sensorIn === undefined || sensorOut === undefined) return <div></div>;
```
Comment (kimibeyl) [inline waterlog/src/Components/Map.js]: Not sure, but do you still have to check for undefined in es6?

Comment (kimibeyl) [reply waterlog/src/Components/Map.js]: and shouldn't undefined be in "", again can't remember

File: waterlog/src/Components/Map.js [hunk 6]
```diff
@@ -1,132 +1,189 @@
...
+    let sensorInColor = defaultColor,
+      sensorOutColor = defaultColor,
+      segmentColor = defaultColor;
+    if (sensorIn.status.toLowerCase() === 'fault') {
```
Comment (kimibeyl) [inline waterlog/src/Components/Map.js:70]: should you test if sensorIn.status exists before checking sensorIn.status.toLowercase?

File: waterlog/src/Components/Map.js (hunk 5 above)
Comment (GiftLanga) [reply waterlog/src/Components/Map.js]: I am trying to check whether a variable has valid data or not.

Comment (GiftLanga) [reply waterlog/src/Components/Map.js]: Is there a better way?

File: waterlog/src/Components/Map.js (hunk 6 above)
Comment (GiftLanga) [reply waterlog/src/Components/Map.js:70]: I used this line "if (sensorIn === undefined || sensorOut === undefined) return <div></div>;" to test if the variable is valid.

File: waterlog/src/Components/Map.js (hunk 5 above)
Comment (kimibeyl) [reply waterlog/src/Components/Map.js]: Like I say not sure, butI think you can just say if(sensorIn || sensorOut). Maybe just make sure in regards to es6. Checkout https://www.reddit.com/r/javascript/comments/58fwyz/good_way_to_check_for_variable_being_not_null_and/

File: waterlog/src/Components/Map.js (hunk 6 above)
Comment (kimibeyl) [reply waterlog/src/Components/Map.js:70]: but not sensorIn.status

Comment (kimibeyl) [reply waterlog/src/Components/Map.js:70]: sensorIn might exist yes, but that does not mean sensorIn.status exist

Comment (GiftLanga) [reply waterlog/src/Components/Map.js:70]: Ohh yah, I see. I will update that

**Review by kimibeyl — COMMENTED** [review]
I realized your team mates code was merged before your PR was merged so you won't be penalized for merge conflicts :)

File: waterlog/src/Components/Map.js (hunk 5 above)
Comment (GiftLanga) [reply waterlog/src/Components/Map.js]: thanks

**Review by kimibeyl — APPROVED** [review]
LGTM

**Review by kimibeyl — APPROVED** [review]

**Pull request merged**